zcutil/distclean.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+# Copyright (c) 2020 The Zcash developers
+
+zcutil/clean.sh
+
+rm -rf depends/*-*-*
+rm -rf depends/work
```

Comment:
While this isn't generated as a result of running the root makefile, it is generated as a result of running a `make` command, so move this to `clean.sh`.